library/java/src/io/envoyproxy/envoymobile/engine/JvmBridgeUtility.java

```diff
@@ -0,0 +1,75 @@
+package io.envoyproxy.envoymobile.engine;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.envoyproxy.envoymobile.engine.types.EnvoyHTTPCallbacks;
+
+/**
+ * Class to assist with passing types from native code over the JNI. Currently supports
+ * HTTP headers.
+ */
+class JvmBridgeUtility {
```

Comment:
I might be missing something, but division into this utility class is not predicated on the returned context, right?

Reply:
No, but it does encapsulate the complexity a bit better (imo), and it allows re-use (in the case of filters).